raiden/network/transport/matrix/transport.py

```diff
@@ -1244,79 +1251,13 @@ def _leave_unused_rooms(self, _address_to_room_ids: Dict[AddressHex, List[_RoomI
         _msg = '_leave_unused_rooms called without account data lock'
         assert self._account_data_lock.locked(), _msg
 
-        # TODO: Remove the next five lines and check if transfers start hanging again
+        # TODO: To be implemented, see https://github.com/raiden-network/raiden/issues/3262
         self._client.set_account_data(
             'network.raiden.rooms',  # back from cast in _set_room_id_for_address
             cast(Dict[str, Any], _address_to_room_ids),
         )
         return
 
-        # cache in a set all whitelisted addresses
```

Comment:
This shouldn't be removed, but instead re-enabled by removing return above. This logic prevents some malicious partner from a DoS path where he triggers us to join and listen on too many rooms, without leaving previous/unused ones. While we use rooms (instead of to device message or other mechanism), we must ensure only a limited number of rooms are kept for each user, which this function tries to do.

Reply:
> This shouldn't be removed, but instead re-enabled by removing return above.

Will do, but will start with tests and increment, as this path caused #3146 and was muted since than - so re-enabling might cause regression.